test/e2e/network/dns_common.go

```diff
@@ -447,10 +447,10 @@ func createProbeCommand(namesToResolve []string, hostEntries []string, ptrLookup
 		}
 		fileName := fmt.Sprintf("%s_udp@%s", fileNamePrefix, name)
 		fileNames = append(fileNames, fileName)
-		probeCmd += fmt.Sprintf(`test -n "$$(dig +notcp +noall +answer +search %s %s)" && echo OK > /results/%s;`, name, lookup, fileName)
+		probeCmd += fmt.Sprintf(`check="$$(dig +notcp +noall +answer +search %s %s)" && test -n "$check" && echo OK > /results/%s;`, name, lookup, fileName)
```

Comment:
I haven't investigated why the `$`s are doubled elsewhere, but do you need to double them in `"$check"` as well?

(Also, an alternative approach would be to just append `2>/dev/null` to the dig command, to discard stderr so you're only checking stdout.)

Reply:
The explanation for $$ is just a few lines above this change. Basically, it says that $(...) is expanded by Kubernetes, which is why $$(...) is used instead. It doesn't say anything about $var_name though. 

But for the 2> /dev/null thing, it won't work, because dig explicitly outputs the error to stdout instead of stderr. Otherwise, this shouldn't have been an issue in the 1st place.

Reply:
For example:

    ubuntu@ubuntu:~$ dig +notcp +noall +answer +search kubernetes.default.svc.cluster.local A
    ;; connection timed out; no servers could be reached
    ubuntu@ubuntu:~$ dig +notcp +noall +answer +search kubernetes.default.svc.cluster.local A 2> /dev/null
    ;; connection timed out; no servers could be reached

Reply:
And as far as $check or $$check goes, it seems that they both behave the same.